src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1999,9 +2006,34 @@ public async Task<JArray> GetHoistedLocalVariables(int objectId, IEnumerable<JTo
                 {
                     var match = regexForAsyncLocals.Match(fieldName);
                     if (match.Success)
+                    {
+                        if (!method.Info.ContainsAsyncScope(Convert.ToInt32(match.Groups[4].Value), offset))
+                            continue;
                         asyncLocal["name"] = match.Groups[1].Value;
+                    }
                     asyncLocalsFull.Add(asyncLocal);
                 }
+                //VB language
+                else if (fieldName.StartsWith("$VB$Local_", StringComparison.Ordinal))
+                {
+                    asyncLocal["name"] = fieldName.Remove(0, 10);
+                    asyncLocalsFull.Add(asyncLocal);
+                }
+                else if (fieldName.StartsWith("$VB$ResumableLocal_", StringComparison.Ordinal))
+                {
+                    var match = regexForVBAsyncLocals.Match(fieldName);
+                    if (match.Success)
+                    {
+                        if (!method.Info.ContainsAsyncScope(Convert.ToInt32(match.Groups[2].Value) + 1, offset))
+                            continue;
+                        asyncLocal["name"] = match.Groups[1].Value;
+                    }
+                    asyncLocalsFull.Add(asyncLocal);
+                }
+                else if (fieldName.StartsWith("$"))
+                {
+                    continue;
+                }
                 else
                 {
                     asyncLocalsFull.Add(asyncLocal);
```

Comment:
Should this be unconditional? It seems to be common, for the cases we don't `continue` earlier.